camkes/templates/dtb-query-common.template.c

```diff
@@ -51,39 +51,57 @@
     /*- do stash('reg_set', reg_set) -*/
 /*- endmacro -*/
 
-/*- macro parse_dtb_node_interrupts(node, max_num_interrupts) -*/
-    /*# This section assumes that the `interrupts` binding's format follow those of the #*/
-    /*# ARM GIC (not v3), i.e. cell 1 = SPI, cell 2 = interrupt number and cell 3 the flag. #*/
+/*- macro parse_dtb_node_interrupts(node, max_num_interrupts, architecture) -*/
     /*- set interrupts = node.get('interrupts') -*/
-    /*- set is_extended_interrupts = False -*/
-    /*- if interrupts is none -*/
-        /*- set interrupts = node.get('interrupts_extended') -*/
-        /*- set is_extended_interrupts = True -*/
-    /*- endif -*/
     /*- set irq_set = [] -*/
-    /*- if interrupts is not none -*/
-        /*- if is_extended_interrupts -*/
-            /*- set num_interrupts = len(interrupts) // 4 -*/
-        /*- else -*/
-            /*- set num_interrupts = len(interrupts) // 3 -*/
+    /*- if 'riscv' in architecture -*/
+    /*# This section assumes that the `interrupts` binding's format contains one interrupt #*/
```

Comment:
I think it would be better to mention that this assumes that the bindings follow that of the SiFive PLIC, instead of saying that the binding's format contains one interrupt number per hardware interrupt.

Reply:
That's a good idea, I wasn't sure if it held for other RISC-V dts files or not. I reworded it though